addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -1218,19 +1218,14 @@ export const formatSelection = (editor, formatName, {applyStyle, formatProps} =
         getDeepRange(editor.editable, { splitText: true, select: true, correctTripleClick: true });
     }
 
-    // Get selected nodes within td to handle non-p elements like h1, h2...
-    // Targeting <br> to ensure span stays inside its corresponding block node.
-    const selectedNodesInTds = [...editor.editable.querySelectorAll('.o_selected_td')]
-        .map(node => closestElement(node).querySelector('br'));
     const selectedNodes = getSelectedNodes(editor.editable)
-        .filter(n => n.nodeType === Node.TEXT_NODE && closestElement(n).isContentEditable && isVisibleTextNode(n));
-    const selectedTextNodes = selectedNodes.length ? selectedNodes : selectedNodesInTds;
+        .filter(n => ((n.nodeType === Node.TEXT_NODE && isVisibleTextNode(n)) || n.nodeName === 'BR') && closestElement(n).isContentEditable);
```

Comment:
There are lots of changes here that, according to the commit message, only pertain to some case with headings. I would appreciate if that were in a separate commit. In particular, I see no test making use of headings so I'm really confused about how it relates to the generic issue of losing the selection when applying a format.

Reply:
- The issue where applying a heading to selected cells caused `getDeepRange` to perform a triple-click, resulting in the loss of one cell from the selection, has been resolved. Since it's no longer an issue, I will update my commit message accordingly.
Code link: https://github.com/odoo/odoo/blob/16.0/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L931
- In this case, when there is text inside table cells and the cells are selected, both `selectedNodesInTds` and `selectedNodes` have values. As a result, `selectedTextNodes` takes the value of `selectedNodes`, causing it to overwrite `selectedNodesInTds`. This leads to the selection being applied only to the text, and the table cells become deselected.
Code link: https://github.com/odoo/odoo/blob/16.0/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L1227
Video: https://www.awesomescreenshot.com/video/33638070?key=b0e68c982d9ed2a09738f5a8a2ba9f37
Should I make seperate commit for the above issue?